### PR TITLE
Server workspace support

### LINKED ISSUE #91: Cannot create a server workspace
I have a question concerning server workspaces in Team Services plugin. As I have gathered from [this issue](https://github.com/Microsoft/vso-intellij/issues/47), by default a local workspace is created, but there are some personal configurations, which allow creating a server workspace. Would you please explain what these configurations are? For me, a workspace is always local and "Location" dropdown is always disabled.
image

### LINKED ISSUE #325: Server workspace support
This is an umbrella issue for server workspace support in Azure DevOps plugin.

Points to consider:
- [x] #91: checking out a repository into a server workspace
- [x] #324: importing of existing server workspace
- [x] working around the read only FS flags in the server workspace
- [x] check whether both classic and reactive clients support main operations on a server workspace
- [x] other issues:
  - [x] #272

For now, the main and first obstacle users encounter when they try to open a server workspace is [this code](https://github.com/microsoft/azure-devops-intellij/blob/5fb7e82c363b2172c0ac62d2e6df338ffefb82d2/plugin/src/com/microsoft/alm/plugin/external/commands/FindWorkspaceCommand.java#L65-L66): we never pass the authentication data over to the client (even if we know it's required), so nothing actually works.

### PR DESCRIPTION
**Pull request by ForNeVeR**
- Closes #91.
- Closes #325.

### CODE REVIEW

File: L2Tests/test/com/microsoft/alm/L2/tfvc/TfvcCheckoutTest.java [hunk 1]
```diff
@@ -3,111 +3,20 @@
...
-        try {
-            // clone it
-            // Everything happens synchronously, so no need to worry
-            model.cloneSelectedRepo();
-
-            // verify that it got cloned
-            File tfvcFolder = new File(tempFolder, Path.combine(workspaceName, TFVC_FOLDER));
-            File tfvcFolderWin = new File(tempFolder, Path.combine(workspaceName, TFVC_FOLDER_WIN));
-            Assert.assertTrue(tfvcFolder.exists() || tfvcFolderWin.exists());
-
```
Comment (ForNeVeR) [inline L2Tests/test/com/microsoft/alm/L2/tfvc/TfvcCheckoutTest.java:101]: The code from this file has been moved into `TfvcCheckoutTestBase`.

File: plugin/src/com/microsoft/alm/plugin/external/visualstudio/VisualStudioTfvcClient.java [hunk 2]
```diff
@@ -83,7 +85,23 @@ public static Path detectClientPath() {
         });
     }
 
-    static List<String> executeClientAndGetOutput(
+    private static List<String> readAsLines(InputStream stream, String label) throws IOException {
```
Comment (mfilippov) [inline plugin/src/com/microsoft/alm/plugin/external/visualstudio/VisualStudioTfvcClient.java:88]: Maybe readLines is better?

File: plugin/src/com/microsoft/alm/plugin/external/visualstudio/VisualStudioTfvcCommands.java [hunk 3]
```diff
@@ -30,6 +32,8 @@
     private static final Pattern WORKFOLD_REPORT_PATTERN = Pattern.compile("^(.*?): (.*)$");
     private static final Pattern WORKFOLD_WORKSPACE_NAME_AND_OWNER_PATTERN = Pattern.compile("^(.*?) \\((.*)\\)$");
 
+    private static final String AUTHENTICATION_ERROR_PREFIX = "TF30063:";
```
Comment (mfilippov) [inline plugin/src/com/microsoft/alm/plugin/external/visualstudio/VisualStudioTfvcCommands.java:35]: If we wanted to handle other errors in future? If yes we should create enumeration or class for error code container and parser?

Comment (ForNeVeR) [reply plugin/src/com/microsoft/alm/plugin/external/visualstudio/VisualStudioTfvcCommands.java:35]: No, I don't think we want to handle any other errors in the future. Our interaction with Visual Studio TFVC client is very limited (we only get the projects' details to import them into our client), and there're no plans to significantly expand it.

**Pull request merged**